### ISSUE TITLE
Not working offline: "Rendering finished, waiting for processing"

### ISSUE THREAD

**Issue by hobbes444** [rwv/lookscanned.io#4]
### Steps to reproduce

1. Load https://lookscanned.io/scan
2. Turn off Internet connection
3. add pdf file
4. Preview of unchanged PDF appears in the left panel
5. Click "Preview" – nothing happens, nothing appears in the right panel.
6. Click "Download": message "Rendering finished, waiting for processing" is displayed, but no pdf is downloaded.

Leaving out step 2 (disabling Internet) works around the issue.

### What was attempted

1. Use Firefox 99 on macOS 12
2. Use Chrome 100 on macOS 12
3. Use private/incognito tab (all add-ons disabled) in the above two configurations.

**Comment by rwv** [comment]
You may need some time for service worker to download all assets. Meanwhile, It seems that the service worker didn't let service-worker activate immediately. Need to hava a look at `vite-paw-plugin`.

**Comment by Iksas** [comment]
Wait for the default PDF to render first, and then turn off your Internet connection. 
After this, I've been able to process my PDFs completely offline.

**Comment by hobbes444** [comment]
Thanks @Iksas, unfortunately this doesn't fix the issue:

1. Load https://lookscanned.io/scan
2. **Wait for default PDF to render in left panel and wait for preview of the result to render in right panel.**
3. Turn off Internet connection
4. add pdf file
5. Preview of unchanged PDF appears in the left panel, nothing appears in the right panel automatically.
6. Click "Preview" – nothing happens, nothing appears in the right panel.
7. Click "Download": message "Rendering finished, waiting for processing" is displayed, but no pdf is downloaded.

**Comment by rwv** [comment]
@hobbes444 @Iksas 
Refresh the page in offline mode and things work fine. Therefore it's something related to service worker however I still didn't figure out why this happens.

**Comment by hobbes444** [comment]
@rwv : you mean hit command+R on a Mac right?

This doesn't change anything for me unfortunately, exact same behaviour afterwards. Tried on Firefox 99 in incognito and normal tab.

**Comment by rwv** [comment]
> @rwv : you mean hit command+R on a Mac right?

@hobbes444 
Yes. Could you please screenshot your Chrome's Application Tab in Devtools?
![image](https://user-images.githubusercontent.com/7891383/164254655-4cfc736d-a3f0-4255-b5df-52383c1980c3.png)

**Comment by hobbes444** [comment]
ha, it works in Chrome after hitting command+R (while offline)! Just not in Firefox.

**Comment by rwv** [comment]
Failed to find out how to enter offline mode in Firefox :(

**Comment by rwv** [comment]
To avoid breaking cache coherence related to service worker, this issue **won't fix**.
Why Firefox does not work still needs some researching.
